auth/server/base/src/main/java/org/wildfly/security/auth/server/jwt/WildFlyElytronTokenProvider.java

```diff
@@ -0,0 +1,603 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.server.jwt;
+
+import static org.wildfly.security.auth.server._private.ElytronMessages.log;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.crypto.SecretKey;
+import javax.security.auth.x500.X500Principal;
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import org.jose4j.jwa.AlgorithmConstraints;
+import org.jose4j.jwe.JsonWebEncryption;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.NumericDate;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwt.consumer.JwtContext;
+import org.jose4j.lang.JoseException;
+import org.wildfly.common.Assert;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.auth.server._private.ElytronMessages;
+import org.wildfly.security.authz.Roles;
+import org.wildfly.security.x500.cert.SelfSignedX509CertificateAndSigningKey;
+import io.smallrye.jwt.algorithm.KeyEncryptionAlgorithm;
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+import io.smallrye.jwt.build.Jwt;
+import io.smallrye.jwt.build.JwtClaimsBuilder;
+import io.smallrye.jwt.util.KeyUtils;
+
+/**
+ * A token provider which holds information regarding dynamic token issuance
+ * including encryption and signing keys. This configuration is mapped to a {@link SecurityDomain}
+ * to use for all tokens generated during the authentication process associated with this Security Domain.
+ *
+ * @author <a href="mailto:szaldana@redhat.com">Sonia Zaldana</a>
+ */
+public class WildFlyElytronTokenProvider implements TokenProvider {
+
+    private KeyStore keyStore;
+    private String issuer = "WildFly Elytron";
+    private long accessTokenExpiryTime = 300;
```

Comment:
For constants like this, it's good to declare it as follows:

`private static final long ACCESS_TOKEN_EXPIRY_TIME = 300;`

Reply:
I was using this variable as a default initial value, although it can be altered with the builder if they choose to use a different expiry time. Should I create some constants and have them be separate instead?

Reply:
Ah, got it. Yes, you can create a separate constant and initialize `accessTokenExpirytTime` using the constant.